ProtoGame2D/Code/Game/Framework/App.cpp

```diff
@@ -74,24 +74,37 @@ void App::BeginFrame()
 
 	m_Shader->Bind();
 
-	m_Shader->SetUniformMat4f("u_Proj", Mat4::orthographic(0.0f, 1024.0f, 0.0f, 1024.0f, -2.0f, 2.0f));
+	Mat4 camera = Mat4::orthographic(0.0f, 1024.0f, 0.0f, 1024.0f, -2.0f, 2.0f);
+
+	m_Shader->SetUniformMat4f("u_Proj", camera);
 }
 
 void App::Update(float deltaseconds)
 {
 	g_InputSystem->Update(deltaseconds);
 	UpdateFromInput();
 	g_InputSystem->EndFrame();
-
-	Renderer::GetInstance()->SwappingBuffers();
 }
 
 void App::Render()
 {
+	Renderer::GetInstance()->DrawQuad(Vec2(200.0f, 200.0f), Vec2(100.0f, 50.0f), Vec4(1.0f, 0.0f, 0.0f, 1.0f), "../Data/Textures/stripes.png", *m_Shader);
```

Comment:
Same as before no hard coded values. Make life easier One variable to rule them all :P